src/craftdef.cpp

```diff
@@ -1107,10 +1119,46 @@ class CCraftDefManager: public IWritableCraftDefManager
 		if (priority_best == CraftDefinition::PRIORITY_NO_RECIPE)
 			return false;
 		if (decrementInput)
-			def_best->decrementInput(input, output_replacement, gamedef);
+			def_best->decrementInput(input, output_replacement, gamedef, operation_count);
 		return true;
 	}
 
+	virtual u16 getCraftOutputCount(const CraftOutput &output) const
+	{
+		// If not otherwise specified, output is 1 of item
+		u16 output_count = 1;
+		// Recipes' quantity is in the string, so we need to split it out
+		std::vector<std::string> split = str_split(output.item, ' ');
+		// If we ended up with more than one element in the vector, get the output quantity from the second element
+		if (split.size() > 1) {
+			output_count = stoi(split[1]);
+		}
+		return output_count;
+	}
+
+	virtual bool setCraftOutputCount(CraftOutput &output, u16 newCount) const
+	{
+		// If the output count isn't specified, it defaults to 1, and isn't included in the string
+		// Otherwise, it's in the second element of a space delimited list
+		std::vector<std::string> split = str_split(output.item, ' ');
```

Comment:
operating on string representations of items is unportable and prone to mistakes
use the `ItemStack` class

Reply:
Thanks for the pointer! I just pushed the changes.